aten/src/ATen/native/sparse/SparseCsrTensorMath.cpp

```diff
@@ -59,6 +59,26 @@ void convert_indices_from_coo_to_csr_cpu(const Tensor& result, const Tensor& inp
     data_out[i] = static_cast<output_t>(numel);
 }
 
+template <typename F, typename ...Args>
+void unary_op_out(F op_out, const Tensor& self, const Tensor& result, Args&&... args) {
+  TORCH_INTERNAL_ASSERT(self.is_sparse_csr());
+  TORCH_INTERNAL_ASSERT(result.is_sparse_csr());
+
+  if (!result.is_same(self)) {
+    if (result.numel() == 0) {
+      at::native::resize_as_sparse_csr_(result, self);
+    }
```

Comment:
```cpp
at::native::resize_output(result.col_indices(), self.col_indices().sizes());
// OR
result.col_indices().resize_(self.col_indices().sizes());
```

don't resize the `result.col_indices()` tensor. Is this a bug, or is it expected? cc: @IvanYashchuk

Reply:
If we write
```
auto col_indices = result.col_indices();
col_indices.resize_(self.col_indices().sizes());
```
We would see that `col_indices` is resized, but `result.col_indices()` would have the original size. To replace `result.col_indices()` we would need to use `set_member_tensors` method (result.unsafeGetTensorImpl()->set_member_tensors(...)`).
It's expected, this issue is related https://github.com/pytorch/pytorch/issues/63549.